src/Symfony/Component/Messenger/Command/ConsumeMessagesCommand.php

```diff
@@ -140,8 +145,13 @@ protected function execute(InputInterface $input, OutputInterface $output): void
             throw new RuntimeException(sprintf('Bus "%s" does not exist.', $busName));
         }
 
+        if (!$this->retryStrategyLocator->has($receiverName)) {
```

Comment:
Shouldn't we decorate the receiver instead of having this logic in the command & worker then? 🤔

Reply:
We probably could, but *should* we? The code reads really clearly inside `Worker`, and my thought is sort of that we're setting out the "core" logic that (unless you really want to) everyone gets. At this point, it includes a lot - event dispatching, retry logic (and a stamp being added for this).

So I guess I would say: someone needs to sell hard on the idea of making it a decorator.

Reply:
Or, to say it differently:

* Do we *really* need this?
* If we do want it, could we only move some parts out (e.g. logging decorator, or event listener, event decorator)  and leave others